server/mdm/apple/profile_matcher.go

```diff
@@ -151,5 +156,17 @@ func (p *profileMatcher) RetrieveProfiles(ctx context.Context, externalHostIdent
 }
 
 func keyForExternalHostIdentifier(externalHostIdentifier string) string {
-	return preassignKeyPrefix + externalHostIdentifier
+	return preassignKeyPrefix + firstNRunes(externalHostIdentifier, preassignKeySuffixMaxLen)
+}
+
+// firstNRunes grabs the first N runes from the provided string
+func firstNRunes(s string, n int) string {
```

Comment:
Just curious for my general edification, is there a reason to prefer for loop approach over something like below?

```
	if len(s) > n {
		return s[:n]
	}
	return s
```

Reply:
great question! the difference is that `range` loops over every rune (unicode code point) instead of each byte, thus the loop option is unicode safe, example here: https://go.dev/play/p/DrtvyLf-Mlv

more in https://go.dev/doc/effective_go#for

> For strings, the range does more work for you, breaking out individual Unicode code points by parsing the UTF-8. Erroneous encodings consume one byte and produce the replacement rune U+FFFD. (The name (with associated builtin type) rune is Go terminology for a single Unicode code point. See [the language specification](https://go.dev/ref/spec#Rune_literals) for details.) The loop
> 
> ```
> for pos, char := range "日本\x80語" { // \x80 is an illegal UTF-8 encoding
>     fmt.Printf("character %#U starts at byte position %d\n", char, pos)
> }
> ```
> 
> prints
> 
> ```
> character U+65E5 '日' starts at byte position 0
> character U+672C '本' starts at byte position 3
> character U+FFFD '�' starts at byte position 6
> character U+8A9E '語' starts at byte position 7
> ```